python/src/wslink/launcher.py

```diff
@@ -156,10 +157,29 @@ def validateKeySet(obj, expected_keys, object_name):
             all_key_found = False
     return all_key_found
 
-# -----------------------------------------------------------------------------
+def checkSanitize(key_pair, sanitize):
+    if not sanitize:
+     return
+    for key in sanitize:
+        if key in key_pair:
+            checkItem = sanitize[key]
+            value = key_pair[key]
+            if checkItem["type"] == "inList":
+                if not value in checkItem["list"]:
+                    logging.warning("key %s: sanitize %s with default" % (key, key_pair[key]));
+                    key_pair[key] = checkItem["default"]
+            elif checkItem["type"] == "regexp":
+                if not "compiled" in checkItem:
+                    checkItem["compiled"] = re.compile("^" + checkItem["regexp"] + "$")
```

Comment:
Should the `^` and `$` be part of our regexp?

Reply:
No, I'd like to enforce that the regexp match the entire string. Otherwise, it can match just the first part of the string and still pass. There is a 'fullmatch()' method in python 3.4, but not 2.7, so this seemed like the best way to enforce it.

Reply:
is the following regexp valid? `^^a-z$$`